### PR TITLE
[REG-2050] fix render pipeline null bug & add test scene for pointer down/up in legacy input

### PR DESCRIPTION
**Pull request by RG-nAmKcAz**
- Fixes a null exception when using the default render pipeline
- Adds a manual test scene for pointer events per the customer sample we received.
  - NOTE: No changes/fixes were made for input... this sample already worked correctly with our recording
  - I also tested this by forcing the editor to run without both input systems loaded (ONLY using the Legacy InputManager)... our SDK spams a lot of errors, but even with that, the clicks still worked
- Adds WASD keyboard test using our recording / playback as well
```csharp
// Button actions ------------------------
button1.OnPointerDownAsObservable().Subscribe(Button1Action);
button2.OnPointerUpAsObservable().Subscribe(Button2Action);
```

https://github.com/user-attachments/assets/fa333b71-ee9d-4cf8-a6d2-fe40a71de7b4

---

Find the pull request instructions [here](https://www.notion.so/regressiongg/Pull-Request-Process-0d57a7eb582a446983edfacafa406f1e?pvs=4)

Every reviewer and the owner of the PR should consider these points in their request (feel free to copy this checklist so you can fill it out yourself in the overall PR comment)

- [ ] The code is extensible and backward compatible
- [ ] New public interfaces are extensible and open to backward compatibility in the future
- [ ] If preparing to remove a field in the future (i.e. this PR removes an argument), the argument stays but is no longer functional, and attaches a deprecation warning. A linear task is also created to track this deletion task.
- [ ] Non-critical or potentially modifiable arguments are optional
- [ ] Breaking changes and the approach to handling them have been verified with the team (in the Linear task, design doc, or PR itself)
- [ ] The code is easy to read
- [ ] Unit tests are added for expected and edge cases
- [ ] Integration tests are added for expected and edge cases
- [ ] Functions and classes are documented
- [ ] Migrations for both up and down operations are completed
- [ ] A documentation PR is created and being reviewed for anything in this PR that requires knowledge to use
- [ ] Implications on other dependent code (i.e. sample games and sample bots) is considered, mentioned, and properly handled
- [ ] Style changes and other non-blocking changes are marked as non-blocking from reviewers

### CODE REVIEW

File: src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/ScreenRecorder.cs [hunk 1]
```diff
@@ -779,7 +779,8 @@ private IEnumerator RecordFrame(bool endRecording = false, bool endRecordingFrom
                             screenSize = new Vector2Int() { x = screenWidth, y = screenHeight },
                             performance = performanceMetrics,
                             pixelHash = pixelHash,
-                            currentRenderPipeline = GraphicsSettings.currentRenderPipeline.GetType().FullName,
+                            // In Unity, if the GraphicsSettings.currentRenderPipeline property is null, the default render pipeline is the Built-in Render Pipeline.
+                            currentRenderPipeline = GraphicsSettings.currentRenderPipeline == null ? null : GraphicsSettings.currentRenderPipeline.GetType().FullName,
```
Comment (RG-nAmKcAz) [inline src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/ScreenRecorder.cs:783]: Handles null `currentRenderPipeline`, which means.. 'default render pipeline' in Unity

File: src/RGUnityBots/Assets/Scripts/LegacyInputTests/LegacyPointerHandlerUniRx.cs [hunk 2]
```diff
@@ -37,26 +36,27 @@ static T GetOrAddComponent<T>(GameObject gameObject)
...
             gameObject.OnPointerDownAsObservable()
-                .SelectMany(_ => this.gameObject.UpdateAsObservable())
-                .TakeUntil(this.gameObject.OnPointerUpAsObservable())
-                .Select(_ => Input.mousePosition)
-                .RepeatUntilDestroy(this) // safety way
                 .Subscribe(
                     x =>
                     {
                         Debug.Log($"{gameObject.name} OnPointerDownAsObservable()");
+                        clickedIndicator.SetActive(true);
```
Comment (RG-nAmKcAz) [inline src/RGUnityBots/Assets/Scripts/LegacyInputTests/LegacyPointerHandlerUniRx.cs:51]: changes this to be verbosely on the down and up separately like the customer sample

File: src/RGUnityBots/Assets/Scripts/LegacyInputTests/LegacyMouseHandler.cs [hunk 3]
```diff
@@ -4,13 +4,12 @@
 
 public class LegacyMouseHandler : MonoBehaviour
 {
+
+    public GameObject clickedIndicator;
+
     public void Start()
     {
         gameObject.OnMouseDownAsObservable()
```
Comment (RG-nAmKcAz) [inline src/RGUnityBots/Assets/Scripts/LegacyInputTests/LegacyMouseHandler.cs:12]: removes the up/down pair and just listens for the down more accurately

File: src/RGUnityBots/Assets/Scripts/LegacyInputTests/LegacyKeyHandlerUniRx.cs [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    {
+        public Image wIndicator;
+        public Image aIndicator;
+        public Image sIndicator;
+        public Image dIndicator;
+
+        public void Start()
+        {
+
+            gameObject.AddComponent<ObservableUpdateTrigger>()
```
Comment (RG-nAmKcAz) [inline src/RGUnityBots/Assets/Scripts/LegacyInputTests/LegacyKeyHandlerUniRx.cs:20]: new test for getting key input during an observable of Update

File: src/RGUnityBots/Assets/Scripts/LegacyInputTests/ButtonHandlerUniRx.cs [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+using UnityEngine.UI;
+
+namespace LegacyInputTests
+{
+    public class ButtonHandlerUniRx : MonoBehaviour
+    {
+        public GameObject clickedIndicator;
+        private void Start()
+        {
+            GetComponent<Button>().OnClickAsObservable().Subscribe(_ =>
```
Comment (RG-nAmKcAz) [inline src/RGUnityBots/Assets/Scripts/LegacyInputTests/ButtonHandlerUniRx.cs:13]: new test for observing Button.OnClick

File: src/RGUnityBots/Assets/AutoDisable.cs [hunk 6]
```diff
@@ -0,0 +1,22 @@
+using UnityEngine;
+
+public class AutoDisable : MonoBehaviour
```
[the remark this answers is not in the document]
Comment (batu) [reply src/RGUnityBots/Assets/AutoDisable.cs:3]: Either way, maybe this should be placed in the Scripts folder?

**Review by batu — APPROVED** [review]
This looks good. A few questions about how we use the RGUnityBots project in this repo.

**Review by abeizer — APPROVED** [review]

File: src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/ScreenRecorder.cs (hunk 1 above)
Comment (RG-nAmKcAz) [reply src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/ScreenRecorder.cs:783]: I debated doing that, but what 'null' means to Unity could very well change.  I decided on writing the actual value.

File: src/RGUnityBots/Assets/AutoDisable.cs (hunk 6 above)
[the remark this answers is not in the document]
Comment (RG-nAmKcAz) [reply src/RGUnityBots/Assets/AutoDisable.cs]: Moved...

This was 'lazy' on my part / Unity is bad at this.  I used create new behaviour on a game object in the Unity IDE to creae this and forgot to go update its location manually

**Pull request merged**